### DISCUSSION TITLE
always get the same width even though I have set fit Addon or not,

### DISCUSSION THREAD

**Discussion by ivannofick** [sourcelair/xterm.js#5050] [Q&A]
I always get the same width even though I have set fit Addon or not, but there is a difference when I set fitAddon when my typing reaches the end of the text it always overlaps

image not use fitaddon
<img width="1209" alt="Screenshot 2024-04-28 at 01 40 55" src="https://github.com/xtermjs/xterm.js/assets/26406777/bd4cb2cc-9539-4eb4-ac69-215f8ec85ee2">

image use fitaddon
<img width="1214" alt="Screenshot 2024-04-28 at 01 41 26" src="https://github.com/xtermjs/xterm.js/assets/26406777/8f58bad6-808e-43e1-bcbf-c5187e74fc5f">

my code for resize screen xtermjs
[https://codesandbox.io/p/sandbox/summer-worker-mt7gtm](https://codesandbox.io/p/sandbox/summer-worker-mt7gtm)

To be honest, I haven't gotten the answer for a year T_T
stack uses next js version 14.0.4.

thanks you before 🙏

**Comment by jerch** [comment]
The fit addon only changes xterm.js' cols * rows metrics to fill the container div properly. You also have to update the size of the PTY to that new size.

**Reply by ivannofick** [reply]
thanks @jerch, solved for me
